src/routers/datatypeRouters/edges/variants_phenotypes.ts

```diff
@@ -87,20 +233,37 @@ async function variantSearch (input: paramsFormatType): Promise<any[]> {
     delete input.max_alt_freq
     delete input.source
   }
+  let phenotypeFilter = ''
+  if (input.phenotype_id !== undefined) {
+    phenotypeFilter = input.phenotype_id as string
+    delete input.phenotype_id
+  }
+  const hyperEdgeFilters = await getHyperEdgeFilters(input)
+  // should preProcessRegionParam in input? it's also in filterStatements
+  return await getHyperedgeFromVariantQuery(routerEdge, preProcessRegionParam(input, 'pos'), phenotypeFilter, hyperEdgeFilters, queryOptions)
+}
+
+// parse p-value filter on hyperedges from input
+async function getHyperEdgeFilters (input: paramsFormatType): Promise<string> {
+  const hyperEdgeFilters = []
+
+  if (input.log10pvalue !== undefined) {
+    hyperEdgeFilters.push(`${routerEdge.secondaryRouter?.getFilterStatements({ log10pvalue: input.log10pvalue }) as string}`)
+    delete input.log10pvalue
+  }
 
-  const studiesFilter = await studySearchFilters(input)
-  return await routerEdge.getSecondaryTargetsFromHyperEdge(preProcessRegionParam(input, 'pos'), input.page as number, '_key', queryOptions, studiesFilter, input.verbose === 'true')
+  return hyperEdgeFilters.join(' and ')
```

Comment:
could this be refactored as:

```
function getHyperEdgeFilters (input: paramsFormatType): string {
  if (input.log10pvalue !== undefined) {
    return (${routerEdge.secondaryRouter?.getFilterStatements({ log10pvalue: input.log10pvalue }) as string)
  }
  
  return []
}
```
?

`getFilterStatements` is not an async method and it returns the final string ready to be used